### ISSUE TITLE
Quarkus path based authentication

### ISSUE THREAD

**Issue by vishalgoel1988** [quarkusio/quarkus#10201]
**Describe the bug**
Hi,

It seems that path based authentication policy does not work. I have following in my application.properties.
#authn
quarkus.http.auth.permission.authenticated.paths=/*
quarkus.http.auth.permission.authenticated.policy=authenticated
#allow /health/* always for probeness
quarkus.http.auth.permission.health.paths=/health/*                            
quarkus.http.auth.permission.health.policy=permit
quarkus.http.auth.permission.health.methods=GET
#authz/policy enforcer
#quarkus.keycloak.policy-enforcer.enable=true

When I am browsing <url>/health/ready, it is still giving me 401. I tried many diff-2 combinations, but nothing worked out.
Though, I use policy enforcer and DISABLE /health/* in policy enforcer, it works out.

**Expected behavior**
/health/* should return with 200 without policy-enforcer too.

**Actual behavior**
/health/* returns 401

**To Reproduce**
1. Create quarkus project with quarkus-keycloak-oidc plugin.
2. Connect project to keycloak.
3. Try /health with below configs.

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.http.auth.permission.authenticated.paths=/*
quarkus.http.auth.permission.authenticated.policy=authenticated
#allow /health/* always for probeness
quarkus.http.auth.permission.health.paths=/health/*                            
quarkus.http.auth.permission.health.policy=permit
quarkus.http.auth.permission.health.methods=GET
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
(Add any other context about the problem here.)

**Comment by sberyozkin** [comment]
@vishalgoel1988 I think it was fixed for 1.6.0, hi Pedro @pedroigor can you check please ?

**Comment by vishalgoel1988** [comment]
Hi,

Is there any expected release time for 1.6.0?

**Comment by sberyozkin** [comment]
`CR1` will be coming shortly, however I'm not 100% sure about the fix. Are you using `quarkus-keycloak-authorization` and would like some resources be public, `/health/*' etc ? If yes then I'm pretty sure it was resolved but you'd likely need to use keycloak-authorization policy, Pedro @pedroigor can confirm

**Comment by vishalgoel1988** [comment]
I have keycloak-authorization-policy in my pom, but I have disabled policy-enforcer in application.properties.
I want to allow /health/* outside of authentication. Obviously it is out of authorization as policy-enforcer is off.

**Comment by sberyozkin** [comment]
`/*` covers `/health/*`, I think a wildcard needs to be replaced with more specific path policies.
The only other alternative is for Quarkus to start sorting path expressions and come up with a  path selection algorithm when multiple path expressions overlap. But the URI space is typically not unlimited for a given service endpoint. So try to avoid using a wildcard...
CC @stuartwdouglas

**Comment by vishalgoel1988** [comment]
Hi,

Acc. to https://quarkus.io/guides/security#matching-multiple-paths-longest-wins, support is already there and it should work,
Also, path can be have dynamic content, so it is not feasible to put exact path.

**Comment by sberyozkin** [comment]
@vishalgoel1988 Yeah, you are right. How do you disable the `keycloak-authorization` ?

**Comment by sberyozkin** [comment]
`#quarkus.keycloak.policy-enforcer.enable=true` does not disable it, it should be `quarkus.keycloak.policy-enforcer.enable=false`

**Comment by vishalgoel1988** [comment]
https://quarkus.io/guides/security-keycloak-authorization#quarkus-keycloak-keycloak-policy-enforcer-config_configuration
is quarkus.keycloak.policy-enforcer.enable not false by default?

**Comment by stuartwdouglas** [comment]
You can't really mix the policy enforcer and fine grained permissions. If you use the policy enforcer it will check every URL, so both the enforcer and the path based security will need to allow if (i.e. a deny from path based security will always deny, but an allow will still be checked by KC).

We could maybe add the options for KC to be a named security policy, so it could be used with path based auth, i.e. instead of quarkus.http.auth.permission.authenticated.policy=authenticated you would have quarkus.http.auth.permission.authenticated.policy=keycloak-policy-enforcer

I don't really know if this is a good idea though, as I thought the intention with the policy enforcer is that it controls everything.

**Comment by vishalgoel1988** [comment]
But I have policy enforcer off in my configs. Are you saying I should not even have authorization extension?

**Comment by stuartwdouglas** [comment]
#quarkus.keycloak.policy-enforcer.enable=true is not off.

**Comment by stuartwdouglas** [comment]
Oh, it is commented out.

**Comment by vishalgoel1988** [comment]
Yes exactly

**Comment by stuartwdouglas** [comment]
Do you have spaces after /health/* in your config file? I just copy/pasted for my test and it failed because it ends with a space and not a *.

**Comment by vishalgoel1988** [comment]
Urghhh....
Yes, this was the space issue. I should have seen it before. However, I think quarkus should take care of this also.
Anyway, thanks for your help.

**Comment by stuartwdouglas** [comment]
The linked pr should fix it
